### PR TITLE
env: switch to oracle-17 jdk

### PR DESCRIPTION
**Pull request by g4s8**
Change source and target version to 17 in `pom.xml`.
Configured `Dockerfile` to use oracle-17 JDK.
Enabled CI builds for 17 JDK.

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -9,29 +9,23 @@ name: Integation tests
...
       - uses: actions/cache@v1
         with:
           path: ~/.m2/repository
-          key: ${{ runner.os }}-jdk-${{ matrix.java }}-maven-${{ hashFiles('**/pom.xml') }}
+          key: ubuntu-latest-jdk-17-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
-            ${{ runner.os }}-jdk-${{ matrix.java }}-maven-
+            ubuntu-latest-jdk-17-maven-
       - run: mvn -B verify -Pitcase
-      # @todo #851:30min Fix integration tests. Integration tests should start Artipie docker
```
Comment (olenagerasimova) [inline .github/workflows/integration-tests.yml:30]: Why this todo is removed? Not all IT are yet rewritten with usage of Artipie docker image, corresponding ticket will be closed.

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@g4s8 thanks, please, check one comment

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (g4s8) [reply .github/workflows/integration-tests.yml:30]: @olenagerasimova this todo is outdated, it says that we need to enable `mvn verify` command, this is why I removed it. Now I added a new todo with a list of tests that should be fixed, see: https://github.com/artipie/artipie/pull/944/commits/75ed0792fbf3b7f41baeea61f2254e5587f17e6e

**Review by olenagerasimova — APPROVED** [review]
@g4s8 thanks

**Pull request merged**